docs/build.md

```diff
@@ -61,6 +62,8 @@ In order to prevent users from triggering `BuildRuns` (_execution of a Build_) t
 | UndefinedParameter | One or many defined `params` are not defined in the referenced strategy. Please ensure that the strategy defines them under its `spec.parameters` list. |
 | RemoteRepositoryUnreachable | The defined `spec.source.url` was not found. This validation only take place for http/https protocols. |
 | BuildNameInvalid | The defined `Build` name (`metadata.name`) is invalid. The `Build` name should be a [valid label value](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set). |
+| SpecEnvNameCanNotBeBlank | Indicates that the name for a user provided environment variable is blank. |
+| SpecEnvValueCanNotBeBlank | Indicates that the value for a user provided environment variable is blank. |
```

Comment:
Does it mean the user won't be able to set an empty environment variable?

Reply:
That is correct. What would be the point of having a blank env var?

Reply:
setting a blank env var in the build obj could serve as explicitly specifying a default and disabling whatever option the env var pertains to

in a way, it also alerts or reminds build run authors of the env var options available, as it seems likely that a build run author would visit the build object as a reference as part of deciding what to set in the build run

especially if administrators decide to restrict what buildrun authors can see within the strategy objects

I think we should allow emtpy env var values out of the gate.  Was there an implementation pain point around allowing that ?

Reply:
updated in next push to allow empty env var values